source/common/quic/active_quic_listener.cc

```diff
@@ -351,11 +353,15 @@ ActiveQuicListenerFactory::createActiveQuicListener(
     EnvoyQuicCryptoServerStreamFactoryInterface& crypto_server_stream_factory,
     EnvoyQuicProofSourceFactoryInterface& proof_source_factory,
     QuicConnectionIdGeneratorPtr&& cid_generator) {
-  return std::make_unique<ActiveQuicListener>(
+  Network::Address::InstanceConstSharedPtr addr =
+      listen_socket->connectionInfoProvider().localAddress();
+  auto listener = std::make_unique<ActiveQuicListener>(
       runtime, worker_index, concurrency, dispatcher, parent, std::move(listen_socket),
       listener_config, quic_config, kernel_worker_routing, enabled, quic_stat_names,
       packets_to_read_to_connection_count_ratio, crypto_server_stream_factory, proof_source_factory,
       std::move(cid_generator), worker_selector_);
+  server_.hotRestart().registerUdpForwardingListener(std::move(addr), worker_index, *listener);
```

Comment:
I'm going to stack this onto #29306 which will make hotRestart() part of quic_context_ rather than server_, but yes, either way it's a reference to a common object.

It looks like these operations are indeed on a worker thread, which I didn't realize - how is posting to the main thread done? Does hot restart even run on the main thread? I think it would be okay to make any thread-resolving part of the work of registerUdpForwardingListener rather than doing it here, anyway. (Thinking if posting isn't easy, mutex-guarding a map would work for simplicity, since after the registration is complete there'll be no contention so it shouldn't be a significant performance burden.)

Reply:
(Holding off on the stacking until the other one lands because doing it now would probably be disruptive to review.)

Reply:
I don't know if hot restart runs on main thread or other stand-alone thread, but definitely not on worker threads.

Dispatcher has an interface `post()` to post callback function to the thread the Dispatcher object runs event loop on, which is thread safe.

Reply:
But the only dispatcher we have access to here is the worker thread's dispatcher, isn't it? So it wouldn't be useful for getting something onto the hot restarter's thread.